src/components/ChoiceGroup/ChoiceGroup.tsx

```diff
@@ -0,0 +1,112 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+import { ChoiceGroup as FabricChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import { join } from '../../util/classNames';
+import Text from '../Text';
+import './ChoiceGroup.css';
+
+export interface ChoiceGroupOption {
+  /**
+   * A required key to uniquely identify the option.
+   */
+  key: string;
+
+  /**
+   * The label for the option.
+   */
+  label: string;
+
+  /**
+   * The sublabel for the option
+   */
+  sublabel?: string;
+}
+
+export interface ChoiceGroupProps extends BaseComponentProps {
+  /**
+   * Items to be presented to the user.
+   */
+  options: ChoiceGroupOption[];
+
+  /**
+   * Label text that will be rendered above the dropdown.
+   */
+  label?: string;
+
+  /**
+   * Intial selected option. If none is provided and selectionOptional is `false`, the first option is used.
+   */
+  initialSelection?: ChoiceGroupOption['key'];
+
+  /**
+   * If set will allow nothing to be selected
+   * default: false
+   */
+  selectionOptional?: boolean;
+
+  /**
+   * Function to call when user changes the selected item.
+   */
+  onChange(key: ChoiceGroupOption['key']): void;
+}
+
+export interface ChoiceGroupState {
+  selectedKey: ChoiceGroupOption['key'] | undefined;
+}
+
+const getLabelWithSubLabel = (label: string, sublabel: string) => {
+  return () => (
+    <span className="ms-Label">
+      <Text bold={true}>{`${label}:`}</Text>
+      <Text>&nbsp;{sublabel}</Text>
+    </span>
+  );
+};
+
+const mapOption = (option: ChoiceGroupOption) => {
+  const text = option.sublabel ? '' : option.label;
+  const onRenderLabel = option.sublabel ? getLabelWithSubLabel(option.label, option.sublabel) : undefined;
```

Comment:
is there a solution where the onRenderLabel method isn't creating a new function on every render cycle? if it's a private method on our ChoiceGroup component then it could access this.props

Reply:
could probably make it a class method